server/src/services/memory.service.ts

```diff
@@ -1,16 +1,84 @@
 import { BadRequestException, Injectable } from '@nestjs/common';
+import { DateTime } from 'luxon';
 import { JsonObject } from 'src/db';
+import { OnJob } from 'src/decorators';
 import { BulkIdResponseDto, BulkIdsDto } from 'src/dtos/asset-ids.response.dto';
 import { AuthDto } from 'src/dtos/auth.dto';
-import { MemoryCreateDto, MemoryResponseDto, MemoryUpdateDto, mapMemory } from 'src/dtos/memory.dto';
-import { Permission } from 'src/enum';
+import { MemoryCreateDto, MemoryResponseDto, MemorySearchDto, MemoryUpdateDto, mapMemory } from 'src/dtos/memory.dto';
+import { OnThisDayData } from 'src/entities/memory.entity';
+import { JobName, MemoryType, Permission, QueueName, SystemMetadataKey } from 'src/enum';
 import { BaseService } from 'src/services/base.service';
-import { addAssets, removeAssets } from 'src/utils/asset.util';
+import { addAssets, getMyPartnerIds, removeAssets } from 'src/utils/asset.util';
+
+const DAYS = 3;
 
 @Injectable()
 export class MemoryService extends BaseService {
-  async search(auth: AuthDto) {
-    const memories = await this.memoryRepository.search(auth.user.id);
+  @OnJob({ name: JobName.MEMORIES_CREATE, queue: QueueName.BACKGROUND_TASK })
+  async onMemoriesCreate() {
+    const users = await this.userRepository.getList({ withDeleted: false });
+    const userMap: Record<string, string[]> = {};
+    for (const user of users) {
+      const partnerIds = await getMyPartnerIds({
+        userId: user.id,
+        repository: this.partnerRepository,
+        timelineEnabled: true,
+      });
+      userMap[user.id] = [user.id, ...partnerIds];
+    }
```

Comment:
AFAICT you aren't using the features of a record here at all. You could just `users.map(async ({id}) => [id, ...await getMyPartnerIds(...)])`